index.js

```diff
@@ -15,11 +15,12 @@ function MultiStream (streams, opts) {
   self._drained = false
   self._forwarding = false
   self._current = null
+  self._toStreams2 = (opts && opts.objectMode) ? toStreams2Obj : toStreams2
 
   if (typeof streams === 'function') {
     self._queue = streams
   } else {
-    self._queue = streams.map(toStreams2)
+    self._queue = streams.map(self._toStreams2)
```

Comment:
`map` passes additional arguments. Specifically, the second argument will be the index of the `streams` array. For the `toStreams2` function, this will cause the `opts` argument to be a number, which is then passed into `new stream.Readable(opts)`. Surprised this didn't cause an error.

Can you fix it so it doesn't try to do that?

Reply:
fixed!